cluster/encode.go

```diff
@@ -0,0 +1,159 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+func (p Params) MarshalJSON() ([]byte, error) {
+	type fieldsOnly Params // Marshal fields-only version of params
```

Comment:
@corverroos what does this renaming of Params?

Reply:
I've refactored this to use json formatter type, same as `Lock`

Reply:
ok, I didn't understood this version